ts/components/conversation/message/message-content/MessageReactBar.tsx

```diff
@@ -51,45 +51,132 @@ const ReactButton = styled.span`
   }
 `;
 
-export const MessageReactBar = (props: Props): ReactElement => {
-  const { action, additionalAction } = props;
-  const [recentReactions, setRecentReactions] = useState<RecentReactions>();
+const StyledContainer = styled.div<{ expirationTimestamp: number | null }>`
+  position: absolute;
+  top: ${props => (props.expirationTimestamp ? '-106px' : '-56px')};
+  display: flex;
+  flex-direction: column;
+  min-width: 0;
+  align-items: flex-start;
+  left: -1px;
+`;
+
+export const MessageReactBar = ({ action, additionalAction, messageId }: Props): ReactElement => {
+  const recentReactions = getRecentReactions();
+  const expirationTimestamp = useIsRenderedExpiresInItem(messageId);
 
-  useMount(() => {
-    const reactions = new RecentReactions(getRecentReactions());
-    if (reactions && !isEqual(reactions, recentReactions)) {
-      setRecentReactions(reactions);
-    }
-  });
+  return (
+    <StyledContainer expirationTimestamp={expirationTimestamp}>
+      <StyledMessageReactBar>
+        {recentReactions &&
+          recentReactions.map(emoji => (
+            <ReactButton
+              key={emoji}
+              role={'img'}
+              aria-label={
+                nativeEmojiData?.ariaLabels ? nativeEmojiData.ariaLabels[emoji] : undefined
+              }
+              onClick={() => {
+                action(emoji);
+              }}
+            >
+              {emoji}
+            </ReactButton>
+          ))}
+        <SessionIconButton
+          iconColor={'var(--emoji-reaction-bar-icon-color)'}
+          iconPadding={'8px'}
+          iconSize={'huge'}
+          iconType="plusThin"
+          backgroundColor={'var(--emoji-reaction-bar-icon-background-color)'}
+          borderRadius="300px"
+          onClick={additionalAction}
+        />
+      </StyledMessageReactBar>
+      <ExpiresInItem expirationTimestamp={expirationTimestamp} />
+    </StyledContainer>
+  );
+};
+
+function formatExpiry({ expirationTimestamp }: { expirationTimestamp: number }) {
+  const diffMs = expirationTimestamp - Date.now();
+  const diff = moment(diffMs).utc();
+
+  if (diffMs <= 0) {
+    return `0s`;
```

Comment:
Shouldn't prettier through a warning and make you change it to `'0s'` since there is no variable in the template string

Reply:
Resolved